File: app/src/main/java/com/babestudios/companyinfouk/ui/filinghistory/FilingHistoryActivityView.java
package com.babestudios.companyinfouk.ui.filinghistory;

import com.babestudios.companyinfouk.data.model.filinghistory.FilingHistoryList;
import com.babestudios.companyinfouk.ui.search.SearchPresenter;

import net.grandcentrix.thirtyinch.TiView;
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread;

interface FilingHistoryActivityView extends TiView {

	@CallOnMainThread
	void showProgress();

	@CallOnMainThread
	void hideProgress();

	@CallOnMainThread
	void showError();

	@CallOnMainThread
	void showFilingHistory(final FilingHistoryList filingHistoryList, FilingHistoryPresenter.CategoryFilter categoryFilter);

	@CallOnMainThread
	String getCompanyNumber();

	@CallOnMainThread
	String getFilingCategory();

	void setFilterOnAdapter(FilingHistoryPresenter.CategoryFilter filterState);

	void setInitialCategoryFilter(FilingHistoryPresenter.CategoryFilter filterState);
}

File: app/src/main/java/com/babestudios/companyinfouk/ui/filinghistory/FilingHistoryActivityView.kt
package com.babestudios.companyinfouk.ui.filinghistory

import com.babestudios.companyinfouk.data.model.filinghistory.FilingHistoryList
import net.grandcentrix.thirtyinch.TiView
import net.grandcentrix.thirtyinch.callonmainthread.CallOnMainThread

interface FilingHistoryActivityView : TiView {

	@get:CallOnMainThread
	val companyNumber: String

	@get:CallOnMainThread
	val filingCategory: String?

	@CallOnMainThread
	fun showProgress()

	@CallOnMainThread
	fun hideProgress()

	@CallOnMainThread
	fun showError()

	@CallOnMainThread
	fun showFilingHistory(filingHistoryList: FilingHistoryList, categoryFilter: FilingHistoryPresenter.CategoryFilter)

	fun setFilterOnAdapter(categoryFilter: FilingHistoryPresenter.CategoryFilter)

	fun setInitialCategoryFilter(categoryFilter: FilingHistoryPresenter.CategoryFilter)
}